nnstreamer/tensor_trainer/tensor_trainer_nntrainer.cc

```diff
@@ -182,7 +182,9 @@ static int nntrainer_model_push_data(const GstTensorTrainerFramework *fw,
   }
 
   data->push_count++;
+  data->count_lock.lock();
   data->queue_count++;
+  data->count_lock.unlock();
```

Comment:
It is strange to protect INC(queue_count) while you don't protect READ(queue_count) or DEC(queue_count) with the lock. 

Please check the critical sections and protected data. Locking queue_count++ only doesn't look meaningful. (and consider using "atomic")

Reply:
PR is changed, Thank you.